### PR TITLE
[librtmp] refactor [curl] Add features: rtmp

### PR DESCRIPTION
**Pull request by talregev**
I use #38269 for refactor the librtmp without update it.
Thanks for @WangWeiLin-MV for his work too.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/librtmp/vcpkg.json [hunk 1]
```diff
@@ -1,15 +1,30 @@
 {
   "name": "librtmp",
-  "version-date": "2019-11-11",
-  "port-version": 4,
+  "version": "2.6",
+  "port-version": 1,
```
Comment (MonicaLiu0311) [inline ports/librtmp/vcpkg.json:4]:
```suggestion
```

port-version:

This is a version set internally by vcpkg. When the upstream version(`"version": "2.6"`) is updated, `port-version` starts counting from 0 (usually this line is removed from 0).

**Comment by dg0yt** [comment]
IMO rtmpdunp / librtmp should not be added to curl.
https://github.com/microsoft/vcpkg/pull/38161#issuecomment-2126243966

**Comment by talregev** [comment]
> IMO rtmpdunp / librtmp should not be added to curl. [#38161 (comment)](https://github.com/microsoft/vcpkg/pull/38161#issuecomment-2126243966)

librtmp is a port in vcpkg.
If it here, I can use it as it.
Also in ubuntu 24.04 (maybe other distro). it compile with librtmp:

```
tal@HP15:~$ curl --version
curl 8.5.0 (x86_64-pc-linux-gnu) libcurl/8.5.0 OpenSSL/3.0.13 zlib/1.3 brotli/1.1.0 zstd/1.5.5 libidn2/2.3.7 libpsl/0.21.2 (+libidn2/2.3.7) libssh/0.10.6/openssl/zlib nghttp2/1.59.0 librtmp/2.3 OpenLDAP/2.6.7
Release-Date: 2023-12-06, security patched: 8.5.0-2ubuntu10.1
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
```

**Comment by dg0yt** [comment]
As written in the other PR: the presence in Ubuntu has little significance. It might be technical debt. We don't have to accept the same technical debt here when upstream says "DONT USE IT ANYMORE".

It is similar to port `kbr5`: You probably don't use it. We can probably assume that it is barely usable, even if it builds.

**Comment by talregev** [comment]
> As written in the other PR: the presence in Ubuntu has little significance. It might be technical debt. We don't have to accept the same technical debt here when upstream says "DONT USE IT ANYMORE".
> 
> It is similar to port `kbr5`: You probably don't use it. We can probably assume that it is barely usable, even if it builds.

If in that case, I vote to delete librtmp from vcpkg. Will you vote the same?

**Comment by dg0yt** [comment]
In fact, delisting `libtrmp` is a possible consequence. But this should be discussed in https://github.com/microsoft/vcpkg/pull/38161.

**Comment by talregev** [comment]
> In fact, delisting `libtrmp` is a possible consequence. But this should be discussed in #38161.

I am still waiting for respond from librtmp. and if they willing to update or maintain their code.

**Comment by MonicaLiu0311** [comment]
We will temporarily convert this PR to draft until #38161 is merged.

**Comment by MonicaLiu0311** [comment]
@talregev Is there any new progress?

**Comment by MonicaLiu0311** [comment]
OK, thanks for your contribution. Closing this PR as no progress seems to have been made. If a new version is released upstream, please ping us to reopen it.

**Pull request closed without merging**